### PR TITLE
feat: update yarn to v2, update typescript

### PR DESCRIPTION
**Pull request by aweiss-dev**
This PR concentrates on workspace/tooling improvements and dependency updates.

The following has been done:

- update yarn to v2 (https://yarnpkg.com/)
- use yarn 2 typescript-plugin (https://yarnpkg.com/features/plugins/#official-plugins)
- update typescript globally to the latest version
- update other dependencies to match typescript (ts-node, karma etc.)
- move some dev-dependencies from the dependencies array in package.json to devDepenendencies 
- fix a lot of missing dependency problems in the packages (yarn2 is more strict)
- fix some ts related errors
- fix most of yarns resolution warnings
- link workspace dependencies accordingly
- remove lerna bootsrap from the pipeline (https://github.com/lerna/lerna/issues/2910)
- remove keyword 'type' from imports, since it was problematic together with ts-node

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ Collection of npm modules created by Wire's web team:
 
 ```bash
 ## Download dependencies and link packages
-yarn && yarn boot
+yarn
```
Comment (atomrc) [inline README.md:52]: does yarn2 provides the exact same feature as lerna bootstrap? Why did we need the bootstrap before?

File: package.json [hunk 2]
```diff
@@ -49,19 +49,13 @@
...
+    "test:all": "yarn dist && lerna run --no-sort --concurrency 8 test"
   },
   "workspaces": {
-    "nohoist": [
-      "**/@jest/**/*",
-      "**/@types/!(long)**/*",
-      "**/emotion-normalize",
-      "**/jest-emotion",
-      "**/react"
-    ],
```
Comment (atomrc) [inline package.json:62]: What was this?

File: packages/copy-config/package.json [hunk 3]
```diff
@@ -1,7 +1,5 @@
 {
-  "bin": {
-    "copy-config": "src/main/cli.js"
```
Comment (atomrc) [inline packages/copy-config/package.json:3]: are you sure about this change? Sounds like a breaking change to me,  doesn't it?

File: README.md (hunk 1 above)
Comment (aweiss-dev) [reply README.md:52]: as far as i understand, lerna bootstrap is from ancient times when npm had no workspaces and yarn didn't exist.
Yarn2 has a very good workspace integration that takes care of the things lerna bootstrap was supposed to do.

File: package.json (hunk 2 above)
Comment (aweiss-dev) [reply package.json:62]: This is yarn hoisting (decides where to place npm modules). Its handled different in yarn2 (take a look into the yarn.yml), so this code is not needed anymore (yarn actually warns you that is deprecated)

File: packages/copy-config/package.json (hunk 3 above)
Comment (aweiss-dev) [reply packages/copy-config/package.json]: good catch, this one was not on purpose

**Review by atomrc — APPROVED** [review]
Super nice, this feels quite clean. 
Thanks man good job

**Pull request merged**